website_snippet_product_category/controllers/website.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2020 Tecnativa - Alexandre Díaz
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import http
+from odoo.http import request
+
+from odoo.addons.website.controllers.main import QueryURL
+
+
+class Website(http.Controller):
+    @http.route(
+        ["/website_sale/render_product_category"],
+        type="json",
+        auth="public",
+        website=True,
+    )
+    def render_product_category(self, template, **kwargs):
+        categories = request.env["product.public.category"].search(
+            [("parent_id", "=", False), ("website_published", "=", True)],
+            order="child_id,sequence DESC",
+        )
+        keep = QueryURL("/shop", category=0)
+        return request.website.viewref(template).render(
+            {"object": categories, "keep": keep}
+        )
```

Comment:
This `viewref` is interesting indeed. I didn't know of that method.

However, digging a little bit, it's better to stick with the good old `request.render`, which renders views lazily by default (thus making it more extendable and performant):

```suggestion
        return request.render(
            template,
            {"object": categories, "keep": keep}
        )
```

Reply:
Can't use 'render' in json request.